### PR TITLE
Can now insert plots directly from a modal

### PR DESCRIPTION
**Pull request by cdricms**
# Purpose

In my opinion it was a bit tedious to create new graphs and a bit slow.
So to resolve this, a modal can be opened (Open command palette > Plot a function) to insert all the necessary values to create a graph. And when submitted it will insert directly where the cursor is in the note.

Screenshot from 2022-08-13 22-13-59
Screenshot from 2022-08-13 22-14-10

> I would like to apologize if the formatting has changed.

### CODE REVIEW

**Comment by leonhma** [comment]
Wow! Great stuff. I'll take some time to test it (probably today) and then release it. Thank you for this PR. That's a great feature 👍

**Comment by cdricms** [comment]
> Wow! Great stuff. I'll take some time to test it (probably today) and then release it. Thank you for this PR. That's a great feature +1

You're welcome and thanks !
Your project has some great potential, especially among university students like me.
So keep up the good work !
Cheers

File: src/PlotFunctionModal.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  static parseToPlot(result: HeaderOptions & { functions: string }): string {
+    return `
+\`\`\`functionplot
+---
+title: ${result.title}
+disableZoom: ${result.disableZoom}
+bounds: [${result.bounds}]
+grid: ${result.grid}
+xLabel: ${result.xLabel}
+yLabel: ${result.yLabel}
```
Comment (leonhma) [inline src/PlotFunctionModal.ts:33]: Should be `result.<field> ?? "\'\'"` for `title`, `xLabel` and `yLabel` respectively.

File: src/main.ts [hunk 2]
```diff
@@ -1,76 +1,97 @@
...
-
-interface HeaderOptions {
-	title: string,
-	disableZoom: boolean,
-	bounds: [number, number, number, number],
-	grid: boolean,
-	xLabel: string,
-	yLabel: string
-}
+import PlotFunMod from './PlotFunctionModal'
```
Comment (leonhma) [inline src/main.ts:10]: Id like this to be the full `PlotFunctionModal` for better clarity.

**Review by leonhma — CHANGES_REQUESTED** [review]
Great! The only thing I found is, that if you dont have a title, or axis label set the modal writes `undefined` for their values. I made a comment about a simple fix. And some minor naming issue but that's basically nothing.

File: src/PlotFunctionModal.ts (hunk 1 above)
Comment (cdricms) [reply src/PlotFunctionModal.ts]: All done in the new commit. `functions` field also had the same issue.

**Review by leonhma — APPROVED** [review]
Just tested it. Works as exspected.

**Pull request merged**